src/core/mount.c

```diff
@@ -270,23 +271,36 @@ static int update_parameters_proc_self_mountinfo(
                 const char *fstype) {
 
         MountParameters *p;
-        int r, q, w;
+        _cleanup_free_ char *w = NULL, *o = NULL;
+        int r, q, v;
 
         p = &m->parameters_proc_self_mountinfo;
 
-        r = free_and_strdup(&p->what, what);
+        if (what) {
+                w = utf8_escape_invalid(what);
```

Comment:
w -> what_escaped and o -> options_escaped please, we can afford the extra characters to make it readable

Reply:
Updated.